code/_core/world/subsystems/generator_power.dm

```diff
@@ -0,0 +1,23 @@
+SUBSYSTEM_DEF(generator_power)
+    name = "Generator Power Subsystem"
+    desc = "The system for turning off/on lights, and deciding if there is station power."
+    priority = SS_ORDER_NORMAL
+
+    var/list/stationside_lights = list()
+    var/station_power = TRUE
+
+/subsystem/generator_power/proc/turn_off_lights()
+    for(var/obj/structure/interactive/lighting/fixture/L in stationside_lights)
```

Comment:
You should turn this into a for(var/k) loop, and then make it so that L = k. For checks like this intrinsically run istype() while looping through a list, which is redundant if your list contains only one type of object, which is a light fixture.